File: 28-find-the-index-of-the-first-occurrence-in-a-string/find-the-index-of-the-first-occurrence-in-a-string.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        n_len = len(needle)
        for i in range(len(haystack)-len(needle)+1):
            cur = haystack[i:i+n_len]
            if cur==needle:
                return i
        return -1

File: 28-find-the-index-of-the-first-occurrence-in-a-string/find-the-index-of-the-first-occurrence-in-a-string.c
int strStr(char* haystack, char* needle) {
    int n_len = strlen(needle);
    int h_len = strlen(haystack);

    if (n_len == 0) return 0;

    for (int i = 0; i <= h_len - n_len; i++) {
        int j;
        for (j = 0; j < n_len; j++) {
            if (haystack[i + j] != needle[j]) {
                break;
            }
        }

        if (j == n_len) {
            return i;
        }
    }


    return -1;
}
